cranelift/codegen/src/context.rs

```diff
@@ -196,10 +196,16 @@ impl Context {
                 self.func.display()
             );
             self.compute_loop_analysis();
-            let mut eg = FuncEGraph::new(&self.func, &self.domtree, &self.loop_analysis, &self.cfg);
-            eg.elaborate(&mut self.func);
+            let mut alias_analysis = AliasAnalysis::new(&self.func, &self.domtree);
+            let mut pass = EgraphPass::new(
+                &mut self.func,
+                &self.domtree,
+                &self.loop_analysis,
+                &mut alias_analysis,
+            );
+            pass.run();
+            log::info!("egraph stats: {:?}", pass.stats);
```

Comment:
Can we package this up in a `self.egraph_pass()` method, similar to other passes, rather than have all the bits of state exposed in our pass-ordering / mid end? I think that would be a bit nicer.

Reply:
Done!